app/src/main/java/com/android/partagix/model/StartOrEndLoanViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package com.android.partagix.model
+
+import androidx.lifecycle.ViewModel
+import com.android.partagix.model.emptyConst.emptyItem
+import com.android.partagix.model.emptyConst.emptyLoan
+import com.android.partagix.model.emptyConst.emptyUser
+import com.android.partagix.model.item.Item
+import com.android.partagix.model.loan.Loan
+import com.android.partagix.model.loan.LoanState
+import com.android.partagix.model.user.User
+import java.util.Date
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class StartOrEndLoanViewModel(private val db: Database) : ViewModel() {
+
+  private val _uiState =
+      MutableStateFlow(StartOrEndLoanUIState(emptyLoan, emptyItem, emptyUser, emptyUser))
+  val uiState: StateFlow<StartOrEndLoanUIState> = _uiState
+
+  fun update(new: StartOrEndLoanUIState) {
+    _uiState.value = new
+  }
+
+  fun onStart() {
+    val loan = _uiState.value.loan
+    val newLoan =
+        loan.copy(
+            state = LoanState.ONGOING,
+            startDate = Date(),
+            endDate = Date(),
```

Comment:
i saw that when you begin your loan you change the end_date of the loan to the actual date of the beginning of the loan which is problematic. If I'm not mistaken the error is on this line